src/controller_examples/rabbitmq_controller/spec/reconciler.rs

```diff
@@ -184,11 +184,96 @@ pub open spec fn reconcile_core(
             (state_prime, req_o)
         },
         RabbitmqReconcileStep::AfterCreateRoleBinding => {
-            let stateful_set = make_stateful_set(rabbitmq);
-            let req_o = Option::Some(APIRequest::CreateRequest(CreateRequest{
-                namespace: rabbitmq.metadata.namespace.get_Some_0(),
-                obj: stateful_set.to_dynamic_object(),
+            let req_o = Option::Some(APIRequest::GetRequest(GetRequest{
+                key: ObjectRef {
+                    kind: StatefulSetView::kind(),
+                    name: rabbitmq.metadata.name.get_Some_0() + new_strlit("-server")@,
+                    namespace: rabbitmq.metadata.namespace.get_Some_0(),
+                }
             }));
+            let state_prime = RabbitmqReconcileState {
+                reconcile_step: RabbitmqReconcileStep::AfterGetStatefulSet,
+                ..state
+            };
+            (state_prime, req_o)
+        },
+        RabbitmqReconcileStep::AfterGetStatefulSet => {
+            if resp_o.is_Some() && resp_o.get_Some_0().is_GetResponse() {
+                let stateful_set = make_stateful_set(rabbitmq);
+                let get_sts_resp = resp_o.get_Some_0().get_GetResponse_0().res;
+                if get_sts_resp.is_Ok() {
+                    // update
+                    let found_stateful_set = StatefulSetView::from_dynamic_object(get_sts_resp.get_Ok_0());
```

Comment:
Can you refactor it to make it look like this https://github.com/vmware-research/verifiable-controllers/pull/146/files#diff-1c325cd2e5693d6b1afd85bfce7464e62f93486d4dc5367b634b9a29a8194a22R145 ?